### ISSUE TITLE
why test result is very different from train an validation result?

### ISSUE THREAD

**Issue by hellozgm** [tsing90/pytorch_semantic_human_matting#14]
问题描述：我训练的是trimap生成网络部分。 为了更好地训练，我在训练过程中加入了验证集，把训练数据10:1比例分为训练集和验证集。验证集是在每个训练epoch之后进行验证。下图是第50个epoch的验证结果，中间是ground truth，右边是验证结果。
<img width="626" alt="微信图片_20190415010053" src="https://user-images.githubusercontent.com/9148863/56096485-5406d380-5f1b-11e9-9e97-8e76e9c74273.png">
然后我把第50个epoch的模型保存，然后用test.py重新加载后对同一张图片进行测试，下面是测试结果。
<img width="215" alt="微信图片_20190415010105" src="https://user-images.githubusercontent.com/9148863/56096492-87496280-5f1b-11e9-9add-09f880b031ed.png">
可以看出来测试结果要比验证集的结果差好多，请问这是什么原因呢？
下面是保存和加载模型的代码段：
保存模型：
`lastest_out_path = "{}/ckpt_{}.pth".format(self.save_dir_model, epoch)`
 `torch.save(model, lastest_out_path)`
加载模型：
`myModel = torch.load(model_path)`
 `myModel.eval()`
 `myModel.to(device)`

**Comment by tsing90** [comment]
please double check your code, there should be typos or mistakes in your code ......

**Comment by wangsssky** [comment]
@hellozgm 问题解决了么？我也遇到了这个问题…

**Comment by hellozgm** [comment]
> @hellozgm 问题解决了么？我也遇到了这个问题…

没有，模型一直不稳定，我已经换到别的平台了

**Comment by wangsssky** [comment]
@hellozgm 多谢～换了什么平台呢？看到还有一个tf的实现，是这个么？换了平台还有这个问题存在么？

**Comment by pourfard** [comment]
@hellozgm which platform did you switch? I also encountered this problem. Results are good while training but at test time have very poor quality.

**Comment by hellozgm** [comment]
> @hellozgm 多谢～换了什么平台呢？看到还有一个tf的实现，是这个么？换了平台还有这个问题存在么？

我换到了mxnet上去了，在gluon-cv里面的model zoo的PSPnet基础上修改的

**Comment by hellozgm** [comment]
> @hellozgm which platform did you switch? I also encountered this problem. Results are good while training but at test time have very poor quality.

I switch to MxNet, and modify the PSPNet in model zoo of gluon-cv. the test result seems good.

**Comment by tsing90** [comment]
what's your train batch size? there is a known issue due to small batch size, when there are batchnorm layers in the model. reference: https://discuss.pytorch.org/t/performance-highly-degraded-when-eval-is-activated-in-the-test-phase/3323
